lib/improver/utilities/spatial.py

```diff
@@ -230,6 +230,32 @@ def calculate_difference(self, cube, coord_axis):
             cube, coord_name, diff_along_axis)
         return diff_cube
 
+    @staticmethod
+    def gradient_from_diff(diff_cube, ref_cube, coord_axis):
+        """
+        Calculate the gradient along the x or y axis from differences between
+        adjacent grid squares.
+
+        Args:
+            diff_cube (Iris.cube.Cube):
+                Cube containing differences along the x or y axis
+            ref_cube (Iris.cube.Cube):
+                Cube with correct output dimensions
+            coord_axis (String):
+                Short-hand reference for the x or y coordinate, as allowed by
+                iris.util.guess_coord_axis.
+
+
+        Returns:
+            gradient (Iris.cube.Cube):
+                A cube of the gradients in the coordinate direction specified.
+        """
+        grid_spacing = np.diff(diff_cube.coord(axis=coord_axis).points)[0]
+        gradient = diff_cube.copy(data=abs(diff_cube.data / grid_spacing))
```

Comment:
Why does this generalised function returned a gradient that has been forced positive? We may need to take the absolute for alphas, but this function may be used for cases where this is not so.

If we don't absolute the values we may need some metadata to indicate the direction of calculation (as with the forward_difference etc. above).